orbit-2-core/src/test/java/com/babylon/orbit2/BaseDslThreadingTest.kt

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright 2020 Babylon Partners Limited
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.babylon.orbit2
+
+import com.appmattus.kotlinfixture.kotlinFixture
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import java.util.concurrent.CountDownLatch
+
+internal class BaseDslThreadingTest {
+
+    companion object {
+        const val EXPECTED_THREAD_PREFIX = "orbit"
+    }
+
+    private val fixture = kotlinFixture()
+
+    @Test
+    fun `reducer executes on orbit dispatcher`() {
+        val action = fixture<Int>()
+
+        val middleware =
```

Comment:
Single line... this seems to be a problem in a few places so not going to add the same comment